openstates/fl/bills.py

```diff
@@ -268,24 +250,179 @@ def scrape_vote(self, bill, chamber, date, url):
             else:
                 break
 
-        totals = re.search(r'(?ms)\s+(\d+)\s+(\d+)\s+.*?TOTALS', text).groups()
+        totals = re.search(
+            r'(?msu)\s+(\d{1,3})\s+(\d{1,3})\s+.*?TOTALS', text).groups()
         yes_count = int(totals[0])
         no_count = int(totals[1])
-        if yes_count != len(votes['yes']) or no_count != len(votes['no']):
-            raise AssertionError(
-                    "Should have {0} yes votes: {1}\n".
-                    format(yes_count, votes['yes']) +
-                    "and {0} no votes: {1}".
-                    format(no_count, votes['no'])
-                    )
         passed = (yes_count > no_count)
         other_count = len(votes['other'])
 
-        vote = Vote(chamber, date, motion, passed, yes_count, no_count,
+        vote = Vote('upper', date, motion, passed, yes_count, no_count,
                     other_count)
         vote.add_source(url)
         vote['yes_votes'] = votes['yes']
         vote['no_votes'] = votes['no']
         vote['other_votes'] = votes['other']
 
+        vote.validate()
+        bill.add_vote(vote)
+
+    def scrape_floor_vote(self, chamber, bill, date, url):
+        (path, resp) = self.urlretrieve(url)
+        text = convert_pdf(path, 'text')
+        lines = text.split("\n")
+        os.remove(path)
+
+        MOTION_INDEX = 4
+        TOTALS_INDEX = 6
+        VOTE_START_INDEX = 9
+
+        motion = lines[MOTION_INDEX].strip()
+        assert motion, "Senate floor vote's motion name appears to be empty"
```

Comment:
Does this only happen for senate votes? Because we run this for both chambers now, right?